nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html

```diff
@@ -0,0 +1,447 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<h2 mat-dialog-title>Provenance Event</h2>
+<div class="provenance-event">
+    <mat-dialog-content>
+        <mat-tab-group>
+            <mat-tab label="Details">
+                <div class="tab-content py-4">
+                    <div class="absolute inset-0 flex gap-x-4">
+                        <div class="w-full">
```

Comment:
we should standardize on how the spacing of these elements is done. currently it seems we are just using `mb-4` here on each child element to give some space between the pieces. I've seen `mb-5` in other places.

I would actually suggest that we use flex gapping here at the parent and not use the margin setting on each element.

```
<div class="w-full flex flex-col gap-y-2">
```

Reply:
My general strategy for spacing is to use flex gap. However, it's not always possible if some of the children are conditional. For instance, in the attributes tab in this dialog. I can however use it here. I tried `gap-y-2` as you suggested but it felt a little tight. I'll make it 3 for now and we can look to adjust more later if needed.

As for the consistency of `4` vs `5`, we should look to settle on something. What makes it a little tricky is that when used in a form with other `mat-form-fields` (which have their own padding/spacing) more or less margin may be needed.